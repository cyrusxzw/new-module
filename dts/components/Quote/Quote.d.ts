import React from 'react';
import type { QuoteProps } from './Quote.types';
declare const Quote: React.ForwardRefExoticComponent<
  QuoteProps & React.RefAttributes<HTMLDivElement>
>;
export { Quote };
