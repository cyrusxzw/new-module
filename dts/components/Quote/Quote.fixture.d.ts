import type { QuoteProps } from './Quote.types';
declare const QuoteFixture: QuoteProps;
export { QuoteFixture };
