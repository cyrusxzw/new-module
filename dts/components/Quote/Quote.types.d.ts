import type { Themes } from '~/types';
declare type QuoteProps = {
  author: string;
  className?: string;
  content: string;
  theme?: Themes;
};
export type { QuoteProps };
