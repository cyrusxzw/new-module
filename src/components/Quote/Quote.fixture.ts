import type { QuoteProps } from './Quote.types';

const QuoteFixture: QuoteProps = {
  author: 'Lorem ipsum',
  content:
    '‘Lorem ipsum dolor sit amet, consectetur adipiscing sit amet ipsum dolor elit’',
};

export { QuoteFixture };
