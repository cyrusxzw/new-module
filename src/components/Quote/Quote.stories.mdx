import { Meta, Story, Canvas, ArgsTable } from '@storybook/addon-docs/blocks';
import * as knobs from '@storybook/addon-knobs';
import { Quote } from './Quote';
import { QuoteFixture } from './Quote.fixture';

<Meta title="Components/Quote" component={Quote} />

# Quote

<Canvas>
  <Story name="Base component">
    {(() => {
      const author = knobs.text('author', QuoteFixture.author, 'Copy');
      const content = knobs.text('children', QuoteFixture.content, 'Copy');
      const theme = knobs.select(
        'theme',
        ['dark', 'light'],
        'dark',
        'Presentation',
      );
      return <Quote author={author} content={content} theme={theme} />;
    })()}
  </Story>
</Canvas>

## API

<ArgsTable of={Quote} />
