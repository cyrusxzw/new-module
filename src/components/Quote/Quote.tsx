import React, { forwardRef } from 'react';
import cx from 'classnames';
import { useThemeContext } from '~/contexts';
import type { QuoteProps } from './Quote.types';
import styles from './Quote.module.css';

const Quote = forwardRef<HTMLDivElement, QuoteProps>(function QuoteRef(
  { author, className, content, theme },
  ref,
) {
  const currentTheme = useThemeContext(theme, 'dark');
  const classSet = cx(styles.base, styles[currentTheme], className);

  return (
    <div className={classSet} ref={ref}>
      <div className={styles.wrapper}>
        <blockquote className={styles.blockquote}>{content}</blockquote>
        <cite className={styles.author}>{author}</cite>
      </div>
    </div>
  );
});

export { Quote };
