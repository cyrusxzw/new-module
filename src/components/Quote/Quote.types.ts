import type { Themes } from '~/types';

type QuoteProps = {
  author: string;
  className?: string;
  content: string;
  theme?: Themes;
};

export type { QuoteProps };
