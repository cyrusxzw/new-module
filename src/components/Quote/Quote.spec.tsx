import React from 'react';
import { render } from '@testing-library/react';
import { Quote } from './Quote';
import { QuoteFixture } from './Quote.fixture';

describe('<Quote />', () => {
  it('should render base component correctly', () => {
    const { container } = render(
      <Quote author={QuoteFixture.author} content={QuoteFixture.content} />,
    );

    expect(container).toMatchSnapshot();
  });
});
