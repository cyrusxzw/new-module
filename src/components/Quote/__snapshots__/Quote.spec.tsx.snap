// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Quote /> should render base component correctly 1`] = `
<div>
  <div
    class="base dark"
  >
    <div
      class="wrapper"
    >
      <blockquote
        class="blockquote"
      >
        ‘Lorem ipsum dolor sit amet, consectetur adipiscing sit amet ipsum dolor elit’
      </blockquote>
      <cite
        class="author"
      >
        Lorem ipsum
      </cite>
    </div>
  </div>
</div>
`;
